downloader/downloader/providers.py

```diff
@@ -195,3 +196,26 @@ def variant_name(variant):
 		variants["source"] = source
 
 		return {name: variant.uri for name, variant in variants.items()}
+
+class LocalProvider(Provider):
+	"""Provider that uses a local m3u8 recording.
+	"""
+	def __init__(self, directory):
+		self.directory = directory
+
+	def get_media_playlist(self, uri, session=None):
+		with open(uri) as f:
+			playlist = hls_playlist.load(f.read().strip(), uri)
+		return playlist
+
+	def get_media_playlist_uris(self, qualities, session=None):
+		if qualities != ["source"]:
+			raise ValueError("Cannot provide non-source qualities")
+
+		files = os.listdir(self.directory)
+
+		for file in reversed(files):
```

Comment:
What's the purpose of checking in specifically reversed order here? To get the "latest" version of the m3u8?

There's no guarentee that the order returned by `os.listdir` corresponds to anything in particular. In practice it will probably be the order files are created in, but it'd be better to explicitly sort by name:
```suggestion
		for file in sorted(files, reverse=True):
```